### PR TITLE
[SDP-960] serve/httphandler: API endpoint to fetch the verification types

### PR DESCRIPTION
**Pull request by ceciliaromao**
### What

API endpoint to get the verification types
GET `receiver/verification-types` 

### Why

Add dropdown for choosing Verification Type when creating new disbursements on FE

### Known limitations

N/A

#### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR does not mix refactoring changes with feature changes (split into two PRs otherwise).
* [x] This PR's title starts with the name of the package, area, or subject affected by the change.

#### Thoroughness

* [x] This PR adds tests for the new functionality or fixes.
* [x] This PR contains the link to the Jira ticket it addresses.

#### Configs and Secrets

* [x] No new CONFIG variables are required -OR- the new required ones were added to the helmchart's [`values.yaml`] file.
* [x] No new CONFIG variables are required -OR- the new required ones were added to the deployments ([`pr-preview`], [`dev`], [`demo`], `prd`).
* [x] No new SECRETS variables are required -OR- the new required ones were mentioned in the helmchart's [`values.yaml`] file.
* [x] No new SECRETS variables are required -OR- the new required ones were added to the deployments ([`pr-preview secrets`], [`dev secrets`], [`demo secrets`], `prd secrets`).

#### Release

* [x] This is not a breaking change.
* [x] **This is ready for production.**. If your PR is not ready for production, please consider opening additional complementary PRs using this one as the base. Only merge this into `develop` or `main` after it's ready for production!

#### Deployment

* [ ] Does the deployment work after merging?

[`values.yaml`]: ../helmchart/sdp/values.yaml
[`pr-preview`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/stellar-disbursement-platform/backend-helm-values
[`dev`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/backend-helm-values
[`demo`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-backend-helm-values
[`pr-preview secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/externalsecrets-common-previews.yaml#L241-L346
[`dev secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/stellar-disbursement-platform-externalsecrets.yaml
[`demo secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-sdp-externalsecrets.yaml

### CODE REVIEW

File: internal/serve/httphandler/receiver_handler_test.go [hunk 1]
```diff
@@ -1616,3 +1616,42 @@ func Test_ReceiverHandler_BuildReceiversResponse(t *testing.T) {
...
+func Test_ReceiverHandler_GetReceiverVerificatioTypes(t *testing.T) {
+	dbt := dbtest.Open(t)
+	defer dbt.Close()
+
+	dbConnectionPool, err := db.OpenDBConnectionPool(dbt.DSN)
+	require.NoError(t, err)
+	defer dbConnectionPool.Close()
+
+	models, err := data.NewModels(dbConnectionPool)
+	require.NoError(t, err)
```
Comment (CaioTeixeira95) [inline internal/serve/httphandler/receiver_handler_test.go:1629]: Why is this necessary in this test? It isn't being used on the handler's function you're testing. Please only use the necessary for the test.

File: internal/serve/httphandler/receiver_handler_test.go [hunk 2]
```diff
@@ -1616,3 +1616,42 @@ func Test_ReceiverHandler_BuildReceiversResponse(t *testing.T) {
...
+	}
+
+	r := chi.NewRouter()
+	r.Get("/receivers/verification-types", handler.GetReceiverVerificatioTypes)
+
+	req, err := http.NewRequest(http.MethodGet, "/receivers/verification-types", nil)
+	require.NoError(t, err)
+
+	rr := httptest.NewRecorder()
+	r.ServeHTTP(rr, req)
```
Comment (CaioTeixeira95) [inline internal/serve/httphandler/receiver_handler_test.go:1643]:
```suggestion
	rr := httptest.NewRecorder()
	req, err := http.NewRequest(http.MethodGet, "/receivers/verification-types", nil)
	require.NoError(t, err)
	http.HandlerFunc(handler.GetReceiverVerificatioTypes).ServeHTTP(rr, req)
```

File: internal/serve/httphandler/receiver_handler_test.go [hunk 3]
```diff
@@ -1616,3 +1616,42 @@ func Test_ReceiverHandler_BuildReceiversResponse(t *testing.T) {
...
+	respBody, err := io.ReadAll(rr.Body)
+	require.NoError(t, err)
+
+	expectedBody := `[
+  "DATE_OF_BIRTH",
+  "NATIONAL_ID_NUMBER",
+  "PIN"
+]`
+
+	assert.Equal(t, expectedBody, string(respBody))
```
Comment (CaioTeixeira95) [inline internal/serve/httphandler/receiver_handler_test.go:1656]:
```suggestion
	resp := rr.Result()

	respBody, err := io.ReadAll(resp.Body)
	require.NoError(t, err)
	defer resp.Body.Close()

	expectedBody := `[
		"DATE_OF_BIRTH",
		"NATIONAL_ID_NUMBER",
		"PIN"
	]`

	assert.Equal(t, http.StatusOK, resp.StatusCode)
	assert.JSONEq(t, expectedBody, string(respBody))
```

File: internal/serve/httphandler/receiver_handler.go [hunk 4]
```diff
@@ -133,3 +133,12 @@ func (rh ReceiverHandler) GetReceivers(w http.ResponseWriter, r *http.Request) {
...
 	httpjson.RenderStatus(w, http.StatusOK, httpResponse, httpjson.JSON)
 }
+
+// GetReceiverVerification returns a list of verification types
+func (rh ReceiverHandler) GetReceiverVerificatioTypes(w http.ResponseWriter, r *http.Request) {
+	httpjson.Render(w, []data.VerificationField{
+		data.VerificationFieldDateOfBirth,
+		data.VerificationFieldNationalID,
+		data.VerificationFieldPin,
+	}, httpjson.JSON)
```
Comment (CaioTeixeira95) [inline internal/serve/httphandler/receiver_handler.go:143]: Could you please move this to a function in the `data` package? Just like the [`GetAllRoles`](https://github.com/stellar/stellar-disbursement-platform-backend/blob/develop/internal/data/roles.go#L30-L37) method. This way, if a new `VerificationField` is added it automatically will be returned here.

**Review by CaioTeixeira95 — APPROVED** [review]
Thank you for working on this. Feel free to merge after addressing the PR comments.

File: internal/serve/httphandler/receiver_handler_test.go (hunk 1 above)
Comment (ceciliaromao) [reply internal/serve/httphandler/receiver_handler_test.go:1629]: I need `dbConnectionPool` and `models` to set the `handler`, isn't it?
